### ISSUE TITLE
Request: webkit2gtk

### ISSUE THREAD

**Issue by MitchWeaver** [sabotage-linux/sabotage#568]
Sorry for this not exactly being an 'issue', rather a request, but it is my main gripe about sabotage. I don't have my browser, (Surf 2.0). Getting webkit2 into sabotage would open up a lot more useability. Basically right now its firefox or dillo.

I've been trying for days to get this beast to compile and I'm hitting a wall. If you think you're up for it, try it yourself.

------------

## Pre-reqs you'll need:
* gstreamer (check my PR)
* gstreamer-plugins-base (check my PR)
* gstreamer-plugins-good (check my PR)
* libsoup (with my changes, check my PR)
* p11-kit ~~(check my PR)~~ *now merged*
* libsecret-1 ~~ (not in butch, [information](http://linuxfromscratch.org/blfs/view/svn/gnome/libsecret.html))

### and these requirements, that are all in butch:
* Cairo
* Enchant
* libwebp
* icu
* ruby
* sqlite

*I may have forgotten some, but they'll be readily apparent in the errors*

----------------

**Current /src/pkg/webkit2gtk:** http://sprunge.us/KVLB

*Problem*: Where I am currently at, cmake is failing to find <stdlib.h> during webkit2gtk's compile.

The full g++ call during this failure can be found here: http://sprunge.us/egci

According to @rofl0r, this is a problem with cmake not cooperating with sabotage. There was a patch he created with an older version of cmake that I've modfied to patch successfully with the newest version. However,  while the patch applies successfully it doesn't seem to work. I don't know why and not sure where to begin debugging it.

**Here is the patch:** http://sprunge.us/HKSF

Good luck. Any information or tips you might have would be appreciated.

**Comment by JillTyrell** [comment]
I've created a branch over at https://github.com/JillTyrell/sabotage/tree/formitch 

It contains a patched cmake, a libsecret and webkitgtk2 recipe, your gstreamer recipes with some slight corrections, as well as two variant recipes for harfbuzz and glib.

libsoup did not require patching, as such you probably should close that PR for it.

The approach with using variant recipes for harfbuzz and glib is rather hackish. For example, you would not want to install glib-python followed by glib, as this could potentially break anything depending on glib-python.

@rofl0r likely has ideas on how this should be addressed.

Regardless, this builds for about an hour before it bombs out with: 

```
In file included from /include/gtk-2.0/gdk/gdkscreen.h:32:0,
                 from /include/gtk-2.0/gdk/gdkapplaunchcontext.h:31,
                 from /include/gtk-2.0/gdk/gdk.h:32,
                 from /include/gtk-2.0/gtk/gtk.h:32,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/JavaScriptCore/../WebCore/platform/gtk/GUniquePtrGtk.h:23,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/build/DerivedSources/ForwardingHeaders/WebCore/GUniquePtrGtk.h:1,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/Shared/NativeWebTouchEvent.h:36,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/UIProcess/WebPageProxy.h:96,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/UIProcess/WebProcessProxy.h:36,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/UIProcess/WebProcessPool.h:45,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/Shared/API/APIURLRequest.cpp:30:
/include/gtk-2.0/gdk/gdktypes.h:114:39: error: conflicting declaration 'typedef struct _GdkDrawable GdkWindow'
 typedef struct _GdkDrawable           GdkWindow;
                                       ^~~~~~~~~
In file included from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WTF/wtf/Platform.h:966:0,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/WebKit2Prefix.h:31,
                 from <command-line>:0:
/src/build/webkitgtk2/webkitgtk-2.16.6/Source/WTF/wtf/glib/GTypedefs.h:104:27: note: previous declaration as 'typedef struct _GdkWindow GdkWindow'
 typedef struct _GdkWindow GdkWindow;
                           ^~~~~~~~~
make[2]: *** [Source/WebKit2/CMakeFiles/WebKit2.dir/Shared/API/APIURLRequest.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /include/gtk-2.0/gdk/gdkscreen.h:32:0,
                 from /include/gtk-2.0/gdk/gdkapplaunchcontext.h:31,
                 from /include/gtk-2.0/gdk/gdk.h:32,
                 from /include/gtk-2.0/gtk/gtk.h:32,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/JavaScriptCore/../WebCore/platform/gtk/GUniquePtrGtk.h:23,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/build/DerivedSources/ForwardingHeaders/WebCore/GUniquePtrGtk.h:1,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/Shared/NativeWebTouchEvent.h:36,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/UIProcess/WebPageProxy.h:96,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/UIProcess/WebPageGroup.h:31,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/Shared/API/c/WKDeprecatedFunctions.cpp:35:
/include/gtk-2.0/gdk/gdktypes.h:114:39: error: conflicting declaration 'typedef struct _GdkDrawable GdkWindow'
 typedef struct _GdkDrawable           GdkWindow;
                                       ^~~~~~~~~
In file included from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WTF/wtf/Platform.h:966:0,
                 from /src/build/webkitgtk2/webkitgtk-2.16.6/Source/WebKit2/WebKit2Prefix.h:31,
                 from <command-line>:0:
/src/build/webkitgtk2/webkitgtk-2.16.6/Source/WTF/wtf/glib/GTypedefs.h:104:27: note: previous declaration as 'typedef struct _GdkWindow GdkWindow'
 typedef struct _GdkWindow GdkWindow;
                           ^~~~~~~~~
make[2]: *** [Source/WebKit2/CMakeFiles/WebKit2.dir/Shared/API/c/WKDeprecatedFunctions.cpp.o] Error 1
make[1]: *** [Source/WebKit2/CMakeFiles/WebKit2.dir/all] Error 2
make: *** [all] Error 2
```

Unfortunately I'm heading off for holiday and do not have more time to spend on this for at least a week. Good luck!

**Comment by rofl0r** [comment]
RIP

maybe it'd be good to get this working anyway, it would be nice to have an alternative to firefox, and firefox may not be viable in the long term anyway due to the upcoming RUST dependency.
